### DISCUSSION TITLE
📎 Adding support for Astro

### DISCUSSION THREAD

**Discussion by anonrig** [biomejs/biome#1254] [Ideas]
### Description

It would be really good to add support for Astro. (https://astro.build)

**Comment by millette** [comment]
I'm guessing this will have to wait for css and html support first, if I read the [Language support Biome page](https://biomejs.dev/internals/language-support/#html-super-languages-support) correctly:

> ## HTML super languages support
> These languages require CSS, HTML and JavaScript parsing to be supported properly. Once those parsers are available, the works around these super languages can start.

Knowing the Biome website is written with Astro, this is a no brainer :-)

**Reply by anonrig** [reply]
I'm OK with linting and parsing the JS/TS parts first.

**Comment by lmammino** [comment]
+1 on this.

I think it would be really useful to have a first simple version of this where only the JS/TS content in the frontmatter is validated/formatted.

**Comment by flaviocopes** [comment]
Second the idea :)

**Comment by unional** [comment]
Currently biome reports some false positives. e.g.: noUnusedVariables

```astro
---
const { title } = Astro.props;
---

<title>{title}</title>
```

**Reply by awhitford** [reply]
I am [using Astro with this configuration guidance](https://biomejs.dev/blog/biome-v1-6/#partial-support-for-astro-svelte-and-vue-files) and am seeing false positives from `noUnusedVariables` and `noUnusedImports` because it isn't scanning the JSX template part of Astro file, so it doesn't understand that these are indeed being used.

I'm looking forward to a fix.

**Comment by Netail** [comment]
Hey!
As of [Biome 2.3](https://biomejs.dev/blog/biome-v2-3/) we support HTML, Astro, Svelte & Vue under the experimental option: `html.experimentalFullSupportEnabled`.
Support is still experimental due to potentially missed edge cases, but definitely worth trying :) 
In case you find any issues please make sure to [create a GitHub issue](https://github.com/biomejs/biome/issues/new/choose) so we can get closer to stable :)

More on HTML-ish languages can be found here: https://biomejs.dev/internals/language-support/#html-super-languages-support
